synapse/handlers/auth.py

```diff
@@ -470,9 +470,7 @@ async def check_ui_auth(
             # authentication flow.
             await self.store.set_ui_auth_clientdict(sid, clientdict)
 
-        user_agent = request.requestHeaders.getRawHeaders(b"User-Agent", default=[b""])[
-            0
-        ].decode("ascii", "surrogateescape")
+        user_agent = request.get_user_agent(" ")
```

Comment:
I think you want this to be an empty string by default?

```suggestion
        user_agent = request.get_user_agent("")
```

Reply:
:facepalm: